### DISCUSSION TITLE
Upload files

### DISCUSSION THREAD

**Discussion by pnewell** [hummingbird-project/hummingbird#235] [Q&A]
What is the best way to handle uploading files to the server? Specifically the use case is for images, I want to avoid storing them in the database directly so am looking to upload them and then to just store a path/url in the database.

**Comment by adam-fowler** [comment]
I would use AWS S3 or a similar service to store your images. 

To upload to S3 your server should return a presigned URL back to your client and the client can do the upload. You can use [Soto](https://github.com/soto-project/soto) to generate URLs to upload to S3. Here is a quick guide to generating presigned URLs https://soto.codes/2020/12/presigned-urls.html

**Reply by clarkezone** [reply]
Does hummingbird support Resumable Uploads in swift NIO as shown here: https://developer.apple.com/wwdc23/10006 ?
